src/app/libp2p_helper/src/codanet.go

```diff
@@ -257,7 +268,7 @@ func (gs *CodaGatingState) InterceptSecured(_ network.Direction, id peer.ID, add
 	// connections in coda are symmetric: if i am allowed to connect to
 	// you, you are allowed to connect to me.
 	remoteAddr := addrs.RemoteMultiaddr()
-	allow = gs.AllowedPeers.Contains(id) || (!gs.DeniedPeers.Contains(id) && !gs.AddrFilters.AddrBlocked(remoteAddr))
+	allow = gs.AllowedPeers.Contains(id) || (!gs.DeniedPeers.Contains(id) && !gs.AddrFilters.AddrBlocked(remoteAddr) && !gs.blockedInternalAddr(remoteAddr))
```

Comment:
As discussed, please remove AddrFilters and blockedInternalAddr from here  -- blockedInternalAddr we don't want to check on incoming ever and AddrFilters we already check on InterceptAccept